### ISSUE TITLE
[Feature]: support external link in NavItemWithLink

### ISSUE THREAD

**Issue by superwushu** [web-infra-dev/rspress#667]
### What problem does this feature solve?

The `link` field in `NavItemWithLink` is always translated to internal URL, i.e. prepending the base URL. This makes it impossible to refer to any external website in navbar. `socialLinks` is a workaround but not very fit the need. E.g. when external URL is wanted in an item in a NavItemGroup.

### What does the proposed API look like?

Like in `to` and `href` in Docusaurus,`NavItemWithLink` may support both internal and external links. For external links, it's also good to have the typcial square-with-arrow icon.

**Comment by superwushu** [comment]
Some correction: I found `link` in `NavItemWithLink` does support external links of `https://...` but does not for `http://`. So, this may be a bug rather than a feature request. But adding a square-with-arrow icon for the external link will be a good improvement.

**Comment by xc2** [comment]
`http://` does be considered as an external link.

could you please check if the `link` is misspelled, such as `http//...` or `http:/...`

**Comment by sanyuan0704** [comment]
I understand the problem. So there are two place related:

- In `@rspress/plugin-auto-nav-sidebar`, when we handled nav config, we should preserve the external link https://github.com/web-infra-dev/rspress/blob/main/packages/plugin-auto-nav-sidebar/src/walk.ts#L146-L157
- In `@rspress/theme-default`, when we render `NavItemWithLink` with external link, it's better to add a link icon in the nav item.

Is there anyone who want to implement them?

**Comment by superwushu** [comment]
> `http://` does be considered as an external link.
> 
> could you please check if the `link` is misspelled, such as `http//...` or `http:/...`

No, it's correctly spelled. I think the issue is in the code below:
```js
export function addLeadingSlash(url: string) {
  return url.charAt(0) === '/' || url.startsWith('https') ? url : `/${url}`;
}
...
export function withBase(url = '/', base = ''): string {
  const normalizedUrl = addLeadingSlash(url);
...
```
A well implemented`isExternalUrl` function is defined but not used in this case.

```js
export function isExternalUrl(url = '') {
  return (
    url.startsWith('http://') ||
    url.startsWith('https://') ||
    url.startsWith('mailto:') ||
    url.startsWith('tel:')
  );
}
```

**Comment by sanyuan0704** [comment]
> > `http://` does be considered as an external link.
> > could you please check if the `link` is misspelled, such as `http//...` or `http:/...`
> 
> No, it's correctly spelled. I think the issue is in the code below:
> 
> ```js
> export function addLeadingSlash(url: string) {
>   return url.charAt(0) === '/' || url.startsWith('https') ? url : `/${url}`;
> }
> ...
> export function withBase(url = '/', base = ''): string {
>   const normalizedUrl = addLeadingSlash(url);
> ...
> ```
> 
> A well implemented`isExternalUrl` function is defined but not used in this case.
> 
> ```js
> export function isExternalUrl(url = '') {
>   return (
>     url.startsWith('http://') ||
>     url.startsWith('https://') ||
>     url.startsWith('mailto:') ||
>     url.startsWith('tel:')
>   );
> }
> ```

I have fixed the problem in #673 .
